ansible-runner/context/app/project/roles/mirror_appliance/tasks/wait-for-quay.yaml

```diff
@@ -1,9 +1,20 @@
-- name: Waiting up to 3 minutes for Quay to become alive at https://{{ quay_hostname }}/health/instance
-  uri:
-    url: "https://{{ quay_hostname }}/health/instance"
-    method: GET
-    validate_certs: no
-  register: result
-  until: result.status == 200
-  retries: 10
-  delay: 30
+- name: Wait for Quay to become alive and handle failure
+  block:
+    - name: Waiting up to 3 minutes for Quay to become alive at https://{{ quay_hostname }}/health/instance
+      uri:
+        url: "https://{{ quay_hostname }}/health/instance"
+        method: GET
+        validate_certs: no
+      register: result
+      until: result.status == 200
+      retries: 10
+      delay: 30
+  rescue:
+    - name: Print debug logs for quay-app in case of failure
+      command: systemctl --user status quay-app.service
+      register: systemctl_status
+      ignore_errors: yes
+
+    - name: Debug systemctl status output
+      debug:
+        var: systemctl_status.stdout_lines
```

Comment:
Can you make sure it exits at the end of this rescue block? Right now it will continue the rest of the tasks since we are handling the error

Reply:
added a fail task: 
```
  - name: Fail the playbook due to Quay not becoming alive
      fail:
        msg: "Quay did not become alive. Check debug logs above for details." 
```